### DISCUSSION TITLE
Need Footer in csv file

### DISCUSSION THREAD

**Discussion by rakeshmahur** [JoshClose/CsvHelper#1489] [Q&A]
I need footer with record count in csv file as below:

applicationid | name | city
1 | rakesh | delhi
2 | jon | london
recordcount | 2

how can i create csv in this format.

**Comment by JoshClose** [comment]
After you're done writing, however you would like to do that, just write another row.

```cs
csv.WriteRecord(footer);
csv.NextRecord();
```

or

```cs
csv.WriteField("recordcound");
csv.WriteField(records.Count);
csv.NextRecord();
```

**Comment by CabaVarga** [comment]
If you are using the pattern from the tutorial:

```csharp
using (var writer = new StreamWriter("path\\to\\file.csv"))
using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))
{    
    csv.WriteRecords(records);
}
```

Then you can write anything to the `writer` stream, before or after accessing `csvWriter`. 
For example:

```csharp
using (var writer = new StreamWriter("path\\to\\file.csv"))
using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))
{    
    csv.WriteRecords(records);
    writer.WriteLine($"recordcount | {records.Count}");
}
```

You need to be careful when you are reading from that file, though, either through `Peek`-ing into the next line before trying to get a record, or by using the `TryGetField` method, or maybe through Exception handling.
